controllers/disruption_controller.go

```diff
@@ -260,6 +260,25 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return ctrl.Result{Requeue: false}, nil
 		}
 
+		// check if we have reached trigger.pods. If not, skip the rest of reconciliation.
+		if instance.Spec.Trigger != nil && instance.Spec.Trigger.Pods != nil {
+			now := metav1.Now()
+			var noPodsBefore metav1.Time
+			if !instance.Spec.Trigger.Pods.NotBefore.IsZero() {
+				noPodsBefore = instance.Spec.Trigger.Pods.NotBefore
+			}
+
+			if instance.Spec.Trigger.Pods.Offset.Duration() > 0 {
+				noPodsBefore = metav1.NewTime(instance.CreationTimestamp.Add(instance.Spec.Trigger.Pods.Offset.Duration()))
+			}
+
+			if now.Before(&noPodsBefore) {
+				r.log.Debugw("requeuing disruption as we haven't yet reached trigger.pods", "trigger.pods", noPodsBefore)
+
+				return ctrl.Result{Requeue: false}, nil
```

Comment:
You might want to specify a requeue time instead.

Reply:
I have personally found that, in practice, the requeue times aren't respected. I see the disruptions we requeue for duration or gc delay to still get reconciled every few minutes. But I'll put in the calculation for appropriate requeue time

Reply:
I think the requeue is respected only if there is no other events on the object.

TBH I have not enough knowledge.

I would only say it gives an hint to what is expected and might reduce the controller (at least we can hope so).